app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessStartDateController.scala

```diff
@@ -0,0 +1,95 @@
+import uk.gov.hmrc.http.InternalServerException
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.config.AppConfig
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.connectors.IncomeTaxSubscriptionConnector
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.forms.BusinessStartDateForm
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.forms.BusinessStartDateForm.businessStartDateForm
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.models.BusinessStartDate
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.services.AuthService
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.utilities.ImplicitDateFormatter
+import uk.gov.hmrc.play.bootstrap.controller.FrontendController
+import uk.gov.hmrc.play.language.LanguageUtils
+
+import scala.concurrent.{ExecutionContext, Future}
+
+
+@Singleton
+class BusinessStartDateController @Inject()(mcc: MessagesControllerComponents, incomeTaxSubscriptionConnector: IncomeTaxSubscriptionConnector,
+                                            authService: AuthService, val languageUtils: LanguageUtils)
+                                           (implicit val ec: ExecutionContext, val appConfig: AppConfig) extends FrontendController(mcc)
+  with I18nSupport with ImplicitDateFormatter {
+
+
+  implicit class FormUtil[T](form: Form[T]) {
+    def fill(data: Option[T]): Form[T] = data.fold(form)(form.fill)
+  }
+
+
+  def view(businessStartDateForm: Form[BusinessStartDate])(implicit request: Request[AnyContent]): Html =
+    uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.views.html.date_of_commencement(
+      businessStartDateForm = businessStartDateForm,
+      postAction = uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.controllers.routes.BusinessStartDateController.submit(),
+      backUrl = backUrl()
+    )
+
+
+  def show(): Action[AnyContent] = Action.async { implicit request =>
+    authService.authorised() {
+      incomeTaxSubscriptionConnector.getSelfEmployments[BusinessStartDate](BusinessStartDateController.businessStartDateKey).map {
+        case Right(businessStartDateData) =>
+          Ok(view(form.fill(businessStartDateData)))
+        case error => throw new InternalServerException(error.toString)
+      }
+    }
+  }
+
+
+  def submit(): Action[AnyContent] = Action.async { implicit request =>
+    authService.authorised() {
+      form.bindFromRequest.fold(
+        formWithErrors =>
+          Future.successful(BadRequest(view(formWithErrors))),
+        BusinessStartDateData =>
+          incomeTaxSubscriptionConnector.saveSelfEmployments[BusinessStartDate](BusinessStartDateController.businessStartDateKey, BusinessStartDateData) map (_ =>
+            Redirect(uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.controllers.routes.BusinessStartDateController.show()))
+      )
+    }
+  }
+
+  def backUrl(): String =
+    appConfig.incomeTaxSubscriptionFrontendBaseUrl + appConfig.feedbackUrl
```

Comment:
```suggestion
    appConfig.incomeTaxSubscriptionFrontendBaseUrl + appConfig.feedbackUrl
```
I don't think this url is correct, we should send the user back to the income sources page on the subscription frontend